pallets/motion/src/lib.rs

```diff
@@ -0,0 +1,161 @@
+#![cfg_attr(not(feature = "std"), no_std)]
+
+#[cfg(test)]
+mod mock;
+#[cfg(test)]
+mod tests;
+
+#[cfg(feature = "runtime-benchmarks")]
+/// Dispatchable function benchmarks.
+mod benchmarking;
+
+/// Dispatchable function weights.
+pub mod weights;
+pub use pallet::*;
+use sp_runtime::DispatchResult;
+pub use weights::WeightInfo;
+
+extern crate alloc;
+use alloc::{boxed::Box, vec};
+
+/// The motion pallet.
+#[frame_support::pallet]
+pub mod pallet {
+	use frame_support::{
+		dispatch::GetDispatchInfo, pallet_prelude::*, traits::UnfilteredDispatchable,
+	};
+	use frame_system::pallet_prelude::*;
+
+	use super::{DispatchResult, *};
+
+	#[pallet::pallet]
+	pub struct Pallet<T>(_);
+
+	/// Configure the pallet by specifying the parameters and types on which it depends.
+	#[pallet::config]
+	pub trait Config: frame_system::Config {
+		/// The runtime event type.
+		type RuntimeEvent: From<Event<Self>> + IsType<<Self as frame_system::Config>::RuntimeEvent>;
+		/// The runtime call type.
+		type RuntimeCall: Parameter
+			+ UnfilteredDispatchable<RuntimeOrigin = Self::RuntimeOrigin>
+			+ GetDispatchInfo;
+		/// Origin that can act as `Root` origin if a collective has achieved a simple majority
+		/// consensus.
+		type SimpleMajorityOrigin: EnsureOrigin<Self::RuntimeOrigin>;
+		/// Origin that can act as `Root` origin if a collective has achieved a super majority
+		/// consensus.
+		type SuperMajorityOrigin: EnsureOrigin<Self::RuntimeOrigin>;
+		/// Origin that can act as `Root` origin if a collective has achieved a unanimous consensus.
+		type UnanimousOrigin: EnsureOrigin<Self::RuntimeOrigin>;
+		/// Type representing the weight of this pallet
+		type WeightInfo: WeightInfo;
+	}
+
+	/// The events that can be emitted.
+	#[pallet::event]
+	#[pallet::generate_deposit(pub(super) fn deposit_event)]
+	pub enum Event<T: Config> {
+		/// A [SimpleMajorityOrigin] motion was executed. [motion_result] contains the call result
+		DispatchSimpleMajority { motion_result: DispatchResult },
```

Comment:
Minor: missing doc comments for struct fields. I would personally add if it were me, but not a blocker.

Reply:
Nice catch, thanks!